### PR TITLE
Optimize EGP Drawing

### PR DESCRIPTION
**Pull request by Denneisk**
Deduplicates a lot of code and especially optimizes EGP HUD drawing.

### CODE REVIEW

**Comment by Denneisk** [comment]
Might be able to do this better by just having `hudPaint` not even draw and instead just steal from the pre-existing RenderTable.

File: lua/entities/gmod_wire_egp_hud/cl_init.lua [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
+end
+
+function ENT:GetEGPMatrix()
+	return Matrix()
+end
 
 local makeArray
 local makeTable
-if (EGP) then -- If the table has been loaded
+if EGP then -- If the table has been loaded
```
Comment (thegrb93) [inline lua/entities/gmod_wire_egp_hud/cl_init.lua:21]: This if check isn't needed nor is the initialize hook. Mind deleting those too?

File: lua/entities/gmod_wire_egp/cl_init.lua [hunk 2]
```diff
@@ -6,61 +6,20 @@ function ENT:Initialize()
 	self.GPU = GPULib.WireGPU( self )
 	self.GPU.texture_filtering = TEXFILTER.ANISOTROPIC
 
-	self.RenderTable = {}
-	self:EGP_Update( EGP.HomeScreen )
+	self.RenderTable = table.Copy(EGP.HomeScreen)
```
Comment (thegrb93) [inline lua/entities/gmod_wire_egp/cl_init.lua:9]: Need `self.NeedsUpdate = true` here too right?

**Comment by thegrb93** [comment]
Looks good, just make sure to test it a bunch. Looks potentially breaking

**Comment by Denneisk** [comment]
Got so confused looking at this thinking this was the object rewrite.

File: lua/entities/gmod_wire_egp_hud/init.lua [hunk 3]
```diff
@@ -19,6 +19,8 @@ function ENT:Initialize()
 	self.Users = {}
 	self.IsEGPHUD = true
 
+	self:SetNWBool("Resolution", false)
```
Comment (thegrb93) [inline lua/entities/gmod_wire_egp_hud/init.lua:22]: Could this be an entity NetworkVar instead of NW? NW tends to be less reliable.

**Review by thegrb93 — CHANGES_REQUESTED** [review]

**Pull request merged**